common/test/java/com/couchbase/lite/DbCollectionsTest.kt

```diff
@@ -50,6 +74,31 @@ class DbCollectionsTest : BaseCollectionTest() {
         scope = baseTestDb.getScope("micro")
         assertEquals(1, scope?.collectionCount)
         assertNotNull(scope?.getCollection("chintz"))
+
+        scope = baseTestDb.getScope("3icro")
+        assertEquals(1, scope?.collectionCount)
+        assertNotNull(scope?.getCollection("chintz"))
+
+
+        scope = baseTestDb.getScope("-micro")
+        assertEquals(1, scope?.collectionCount)
+        assertNotNull(scope?.getCollection("chintz"))
+
+        //Illegal chars
+        baseTestDb.createCollection("chintz", "!micro")
+    }
+
+    @Test
+    fun testScopeNameCaseSensitive(){
+        baseTestDb.createCollection("coll1", "scope1")
+        baseTestDb.createCollection("coll2", "Scope1")
+
+        val scope1 = baseTestDb.getScope("scope1")
+        val scope2 = baseTestDb.getScope("Scope1")
+
+        assertNotNull(scope1)
+        assertNotNull(scope2)
+        assertNotSame(scope1, scope2)
     }
```

Comment:
You need to add to this test to verify that baseTestDb.getScope("scope1").equals(baseTestDb.getScope("scope1"))